### ISSUE TITLE
A binary or executable for the project

### ISSUE THREAD

**Issue by ghost** [philbot9/youtube-comment-scraper#2]
Hi there,
Can you release a binary version or perhaps an executable of this software?
Thank you for your hard work, your software really works, but still I am hoping of a stand-alone version of it.

**Comment by philbot9** [comment]
Hi,
That's an interesting idea. I can see writing a command line tool in NodeJS that does the same thing. Something like
```
$ youtube-comment-scraper --format=csv --output=/path/to/file.csv [video ID]
```

Would that serve your needs?

**Comment by ghost** [comment]
- Can you release perhaps a stand-alone binary in exe or java, it would be great to have an All-in-one package with all the dependencies which can be used on Windows without the need to install python and the required dependencies.
- You also need to work on the results structure, and make it more presentable and structured, since the current formats, both csv and json, cant present the data in an understandable or organized way, and its difficult to extract meaning, it's very bad trust me.

I think you should consider saving the comments as a Database, by doing-so, we can view the comments and choose at which Field "number, date, user, likes, etc......" to focus on, while maintaining the replies and other fields structure, since a database is interconnected.

Basically, you should focus on creating a Stand-alone version, and more importantly represent the scrapped comments as a Database.

**Comment by ethus3h** [comment]
@christophered:

1. Windows is a proprietary system. I wouldn't expect package maintainers to support proprietary systems unless they happen to require them themselves.
2. No one "need"s to work on anything unless you have some sort of support contract. If you don't like the result structure, you're free to implement it and open a PR, or to ask nicely ("nicely" being the operative word). Complaining doesn't help your case.
3. CSV and JSON are both structured data formats. They are open, simple, easy-to-use formats for the exact purpose you're talking about.

In summary, none of your complaints have any merit whatsoever.

(I'm not affiliated with this project, just thought I'd reply.)

**Comment by ghost** [comment]
@ethus3h 
- I was not complaining about the software, I was suggesting to the developer to focus on improving the structure of the result to be more presentable as a database.
-At the end of the day it works, I said that at the beginning of the topic is that the software is working excellently and gets the job done, his work is appreciated.

Calm down dear, no one is offending your friend, it's just a suggestion to improve and better promote the code.
ps: I will close the topic later.

**Comment by philbot9** [comment]
@christophered Thanks for your feedback. I'm currently compiling the requirements for version 2 of the scraper, so any suggestions are welcome.

- Currently, I'm not planning to create a bundled executable. What I proposed is a command line utility that runs locally on the NodeJS runtime. You could use this on Windows (and most other platforms) as long as you install NodeJS, which is no more complicated than installing the Java runtime.
Using the hosted service (http://ytcomments.klostermann.ca/) doesn't require you to install any dependencies. Could you explain how the website is lacking and why an executable would be better? I'm having a hard time seeing the advantage of a stand-alone binary.

 - The result-structure is deliberately kept simple. @ethus3h is right in that both JSON and CSV are simple-to-use and serve most users' needs. Importing them into a database shouldn't be too hard. [MongoDB](https://www.mongodb.com/what-is-mongodb) is a JSON document database with a powerful query language and it happily imports raw JSON data. If you're looking for something simpler, I believe Excel allows you to export CSV data into a Microsoft Access database (yuck! :wink:).
I wrote this tool specifically for a (non-technical) data researcher and language analyst and so the results are geared towards that use case. Can you elaborate on why "it's very bad" for your use case? Maybe the results can be improved in the next version.

Cheers

**Comment by ghost** [comment]
the command line utility is excellent.
All is good, thank you

**Comment by philbot9** [comment]
Alright then. I've created an empty repo for the CLI and I will post here once I've had time to implement it.

https://github.com/philbot9/youtube-comment-scraper-cli

**Comment by philbot9** [comment]
Version 1 of the command line utility is ready and has been published to npm.

https://www.npmjs.com/package/youtube-comment-scraper-cli

https://github.com/philbot9/youtube-comment-scraper-cli

**Comment by ghost** [comment]
Thanks
